lib/customer.rb

```diff
@@ -0,0 +1,43 @@
+require 'csv'
+
+class Customer
+
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  #### WAVE 2 ####
+
+  def self.all
+
+    all_customers = []
+
+    customer_array = CSV.read('data/customers.csv')
+    customer_array.each do |customer|
+
+      address = {}
+
+      id = customer[0].to_i
+      email = customer[1]
+      address[:street] = customer[2]
+      address[:city] = customer[3]
+      address[:state] = customer[4]
+      address[:zip] = customer[5]
+
+      all_customers << Customer.new(id, email, address)
+
+    end
+    return all_customers
+  end
+
+  def self.find(id)
+    customer = self.all.find { |customer| customer.id == id }
+    return customer
```

Comment:
You can directly return the result of `find`:

```suggestion
    return self.all.find { |customer| customer.id == id }
```